articles/101_build_tool.md

```diff
@@ -0,0 +1,213 @@
+
+### Environment Setup
+
+A very important part beside the actual build of a package is the environment setup.
+For a CMake package e.g. a dependency discovered using the CMake function `find_package` needs to be either in a common location or the location must be provided through the environment variable `CMAKE_PREFIX_PATH`.
+On the other hand after a package has been built and installed the environment might need to be extended to use the package.
+E.g. when a package installs a shared library in a non-default location the environment variable `LD_LIBRARY_PATH` (or `PATH` on Windows) needs to be extended to include the containing folder.
+
+The functionality to setup these environment variables can be provided by either the build tool or the build system.
+In the latter case the build tool only needs to know how the build system exposes the environment setup in order to reuse it.
+
+Considering the use case that a user might want to invoke the build system of each package manually it is beneficial if the build system already provides as much of the environment setup as possible.
+That avoids forcing the user to manually take care of the environment setup when not using a build tool.
+
+## Existing Build Systems
+
+In the following the build systems being used in the ROS ecosystem are briefly described.
+
+### catkin
+
+[catkin](http://wiki.ros.org/catkin) is based on CMake and provides a set of convenience functions to make writing CMake packages easier.
+It automates the generation of CMake config files as well as pkg-config files.
+It additionally provides functions to register different kinds of tests.
+
+### ament_cmake
+
+[ament_cmake](https://github.com/ament/ament_cmake) is an evolution of `catkin` and is also based on CMake.
+The main difference between `ament_cmake` and `catkin` is described in [another article](http://design.ros2.org/articles/ament.html).
+In the context of the build tool the biggest difference is that `ament_cmake` generates package-specific files to setup the environment to use the package after it has been built and installed.
+
+### Python setuptools
+
+`setuptools` is the common tool to package Python packages.
+A Python package uses a `setup.py` file to describe the dependencies as well as how and what to build and install.
+In ROS 2 a package can be a "vanilla" Python package whereas in ROS 1 any Python functionality is triggered from a CMake file.
+
+## Existing Build Tools
+
+Several different build tools are already being used in the ROS ecosystem.
+Their method of operating is being described in the following subsections together with their advantages as well as disadvantages.
+
+### catkin_make
+
+`catkin_make` is provided by the ROS package `catkin` which contains the build system for ROS 1.
+It has been designed as the successor of `rosbuild` for ROS Fuerte.
+
+The tool invokes CMake only a single time and uses CMake's `include` function to process all packages in a single context.
```

Comment:
Technically, doesn't it use the `add_subdirectory` function? It doesn't really matter I think, but as long as you mention it, then we might as well try to be precise.